src/base-client/index.ts

```diff
@@ -12,7 +12,7 @@ export abstract class Client {
   protected generateUrl(basePath: string, path: string, pathParams: PathParams = {}): string {
     const cleanPath = path ? `/${path.replace(/^[\/]/, '')}` : '';
     let baseUrl = `${basePath.replace(/[\/]$/, '')}${cleanPath}`;
-    const pendingParams = baseUrl.match(/{.*}/g) ?? [];
+    const pendingParams = baseUrl.match(/{(.*?)}/g) ?? [];
```

Comment:
Can you add a test for this? It can be under `client.test.ts`, just modify or add a test using 2 or more path params